packages/open-truss/src/configuration/engine-v1.tsx

```diff
@@ -25,23 +29,35 @@ export interface WorkflowV1 {
 export function engineV1(
   COMPONENTS: COMPONENTS,
   config: WorkflowV1,
-  data: YamlType,
 ): RenderingEngine {
-  const renderFrames = (frames: FrameV1[]): ReactTree => {
-    return frames.map(({ view, data, frames: subFrame }, i) => {
-      const { component, props } = view
-      const Component = COMPONENTS[component]
-      return subFrame === undefined ? (
-        <Component key={i} data={data} config={config} {...props} />
-      ) : (
-        <Component key={i} data={data} config={config} {...props}>
-          {renderFrames(subFrame)}
-        </Component>
+  const renderFrames = async (frames: FrameV1[]): Promise<ReactTree> => {
+    return (async () => {
+      return Promise.all(
+        frames.map(async ({ view, data, frames: subFrame }, i) => {
+          const { component, props: viewProps } = view
+          const Component = COMPONENTS[component]
+          const props = {
+            key: i,
+            data,
+            config,
+            ...viewProps,
+          }
+
+          if (subFrame === undefined) {
+            return await Component({ ...props })
+          } else {
+            const subFrames = (await renderFrames(subFrame)).map((child, k) => {
+              return <div key={k}>{child}</div>
```

Comment:
Did you add a div to add a key? To not add a div you can use the long form for a fragment: <React.fragment key={k}>...</React.fragment>

Reply:
Oh, good to know! I'll add that to my todo to fix.